Tests/URLSessionDecodableTests/ParametersTests.swift

```diff
@@ -16,6 +16,29 @@ final class ParametersTests: XCTestCase {
         }
     }
 
+    func testURLEncodingWithExistingParameters() {
+        let params = [
+            "param1": "1",
+            "param2": "abc"
+        ]
+        let request = URLRequest(url: URL(string: "www.viacom.com/test?param1=0&param3=foo")!)
+        let encodedUrl = URLParametersEncoder(parameters: params).encode(into: request).url!
+        let queryItems = URLComponents(url: encodedUrl, resolvingAgainstBaseURL: false)!.queryItems!
+
+        let expectedParameters: [String: String] = [
+            "param1": "1",
+            "param2": "abc",
+            "param3": "foo"
+        ]
+        let urlParameters = queryItems.reduce([String:String]()) { dict, param in
```

Comment:
You could use `reduce(into::)` here instead
https://developer.apple.com/documentation/swift/array/3126956-reduce